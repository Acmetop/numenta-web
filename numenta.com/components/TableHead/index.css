/**
 * Numenta.com company website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.tableHeader {
  /* */
}
