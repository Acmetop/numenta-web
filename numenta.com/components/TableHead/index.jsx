import React from 'react'

import styles from './index.css'


const TableHeader = ({children}) => (
  <thead className={styles.tableHeader}>
    {children}
  </thead>
)

TableHeader.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default TableHeader
