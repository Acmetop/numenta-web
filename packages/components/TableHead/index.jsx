// Numenta shared web components
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const TableHeader = ({border, children}) => {
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {border})
  )

  return (
    <thead className={styles.tableHeader}>
      {childrenWithProps}
    </thead>
  )
}

TableHeader.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node.isRequired,
}

export default TableHeader
