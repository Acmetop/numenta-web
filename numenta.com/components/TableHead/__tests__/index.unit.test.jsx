import React from 'react'
import renderer from 'react-test-renderer'

import TableHead from '../../TableHead'


describe('<TableHead /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <TableHead>
        Table Heading
      </TableHead>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
