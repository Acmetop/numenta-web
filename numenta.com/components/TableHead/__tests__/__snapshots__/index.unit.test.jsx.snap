exports[`TableHead React component Renders correctly 1`] = `
<thead
  className="tableHeader">
  <tr
    className="tableRow">
    <th
      className="tableTitle"
      colSpan={1}>
      Table Title 1
    </th>
    <th
      className="tableTitle"
      colSpan={1}>
      Table Title 2
    </th>
  </tr>
</thead>
`;
